### PR TITLE
WEB-9787 add filters to the queryString

### PR DESCRIPTION
**Pull request by t1gr0u**
#### What are the links to relevant tickets?
[WEB-9187](https://hxshortbreaks.atlassian.net/browse/WEB-9187)

#### What does this PR do?
Added filter for the linked resources so when building the href of that resource we can have filters.

#### What functional/unit tests does this PR have?
None at the mo, but will come later to update the code and add unit test.

#### Is there anything a developer should be aware of when reviewing this?
Get the Works and checkout `WEB-9187` branch, then try:
`http://localhost:3000/hotelProducts/WARMER?include=faqs`

in the linked resources you should have a faqs href like: 
`/faqs/?filter%5BproductIds%5D=WARMER`

#### Screenshots (if appropriate)

---
#### Quality checklist (for PR author to complete BEFORE code review):
- [x] I've checked there is appropriate unit test coverage.
- [x] I've updated documentation with any changes to procedures.
- [x] I've checked this work against the requirements of the Jira.
- [x] This change passes all necessary tests (cross browser, automated or otherwise).
- [x] I am ready for this to be code reviewed, merged and tested.

#### Reviewer 1 @philmeehan
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

#### Reviewer 2
- [ ] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [ ] I’ve checked for appropriate test coverage.
- [ ] I’ve checked for coding anti-patterns.
- [ ] I've checked this work against the requirements of the Jira.
- [ ] I don’t believe this work introduces unnecessary technical debt.
- [ ] +1 from me!

### CODE REVIEW

File: lib/pluginJsonapi.js [hunk 1]
```diff
@@ -158,18 +186,38 @@ function pluginJsonapi( server, options, next ) {
...
-              resource.href = _buildHref( request.route.settings.bind.resourceName, resource.id );
+              resource.href = _buildHref( { type: request.route.settings.bind.resourceName, ids: resource.id } );
             }
+
             // If there's no links then there's no actions to take
             if ( resource.links ) {
+
               // Loop each resource in the links and where we have a 'type' and 'id', build a href
               _.each( resource.links, function( value ) {
+                var requestData = {
```
Comment (kevinhodges) [inline lib/pluginJsonapi.js]: I don't think `requestData` is quite the right name here, what is this variable actually for?

File: lib/pluginJsonapi.js [hunk 2]
```diff
@@ -158,18 +186,38 @@ function pluginJsonapi( server, options, next ) {
...
+                  type: value.type
+                };
+
                 // Need both type and ids to do anything meaningful here
-                if ( value.type && _.size( value.ids ) ) {
+                if ( value.type && ( _.size( value.ids ) || _.size( value.filter ) ) ) {
```
Comment (kevinhodges) [inline lib/pluginJsonapi.js]: we've lost some logic here

File: lib/pluginJsonapi.js [hunk 3]
```diff
@@ -25,31 +25,48 @@ function pluginJsonapi( server, options, next ) {
...
+          result[filter] = filterValues.join( ',' );
+        }
+        return result;
+      }, {} );
+    }
+
+    // get all the query object into a querystring using qs
+    queryString = qs.stringify( query );
+
+    if ( _.isUndefined( ids ) || (_.isString( ids ) && ids === '' ) ) {
```
Comment (philmeehan) [inline lib/pluginJsonapi.js]: Do you need the _.isString here? ids === '' will detect if it is a string and empty as well

File: lib/pluginJsonapi.js [hunk 4]
```diff
@@ -158,18 +194,38 @@ function pluginJsonapi( server, options, next ) {
           _.each( resourceArray, function( resource ) {
             // build a href for the resource if there's an id property
             if ( resource.id ) {
-              resource.href = _buildHref( request.route.settings.bind.resourceName, resource.id );
+              resource.href = _buildHref( { type: request.route.settings.bind.resourceName, ids: resource.id } );
```
Comment (philmeehan) [inline lib/pluginJsonapi.js]: can you break this into multiple lines now it is an object?

File: lib/pluginJsonapi.js [hunk 5]
```diff
@@ -158,18 +194,38 @@ function pluginJsonapi( server, options, next ) {
...
               _.each( resource.links, function( value ) {
+                var hrefParameters = {
+                  type: value.type
+                };
+
                 // Need both type and ids to do anything meaningful here
-                if ( value.type && _.size( value.ids ) ) {
+                if ( value.type && ( _.size( value.ids ) || _.size( value.filter ) ) ) {
+
+                  if ( _.size( value.ids ) ) {
```
Comment (philmeehan) [inline lib/pluginJsonapi.js]: not too sure I like the use of _.size here.

Would !_.isEmpty work in its place?

File: lib/pluginJsonapi.js (hunk 2 above)
Comment (philmeehan) [inline lib/pluginJsonapi.js]: have we? trying to work out what

File: lib/pluginJsonapi.js (hunk 5 above)
Comment (t1gr0u) [inline lib/pluginJsonapi.js]: I will change that when I update the code.

**Pull request merged**